src/Framework/TargetSkippedEventArgs.cs

```diff
@@ -7,6 +7,37 @@
 
 namespace Microsoft.Build.Framework
 {
+    /// <summary>
+    /// A reason why a target was skipped.
+    /// </summary>
+    public enum TargetSkipReason
+    {
+        /// <summary>
+        /// The target was not skipped or the skip reason was unknown.
```

Comment:
A target that was not skipped should never issue a TargetSkippedEvent right? So I'm confused how None can represent a target that was not skipped.

Reply:
Yes in theory None should never be used. But I decided to leave a "null" value just in case... Maybe it should be called Invalid?

Reply:
What about just deleting it altogether? Or asserting that it's never None?

Reply:
Yes I'm guessing I'll just delete it. Felt like if we didn't have a "null" value we could paint ourselves in the corner, e.g. when reading 0 from a deserializer it would map to an invalid value, thus signaling to us that it's an error, but if we don't have a default value and 0 will just map the first field, then we'll never know it was bad. I'm just hypothesizing here.

Reply:
This is the scenario I'm talking about here: 
https://github.com/dotnet/msbuild/blob/454959c05cc1f202b9db459356a9b8d747a0301f/src/Build/Logging/BinaryLogger/BuildEventArgsReader.cs#L300

For older versions we need some default value that represents "null" and I don't want to make it nullable because it'd increase bloat.

Reply:
Feels like we do need a value for None/Unknown/Invalid/Unspecified

Reply:
Yea, and always error via ErrorUtilities.VerifyThrow when a None value flows into a field via the TaskSkippedEventArgs constructor?

Reply:
I added a check